mdata/store_cassandra.go

```diff
@@ -435,12 +449,20 @@ func (c *CassandraStore) SearchTable(key, table string, start, end uint32) ([]ch
 	results := make(chan outcome, numQueries)
 	for i := range crrs {
 		crrs[i].out = results
-		c.readQueue <- crrs[i]
+		select {
+		case c.readQueue <- crrs[i]:
+		default:
+			cassReadQueueFull.Inc()
+			return nil, errReadQueueFull
+		}
```

Comment:
FTR: after discussing with the 3 of us we decided it's better to provide consistent performance and sometimes drop reads that could otherwise have been served if no other tenants were exerting much load